spec/intl.html

```diff
@@ -2148,13 +2192,24 @@ <h1>Temporal.Calendar.prototype.mergeFields ( _fields_, _additionalFields_ )</h1
             1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
             1. Set _fields_ to ? ToObject(_fields_).
             1. Set _additionalFields_ to ? ToObject(_additionalFields_).
-            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
-              1. Return ? DefaultMergeCalendarFields(_fields_, _additionalFields_).
-            1. Let _fieldsCopy_ be OrdinaryObjectCreate(%Object.prototype%).
-            1. Perform ? CopyDataProperties(_fieldsCopy_, _fields_, « », « *undefined* »).
-            1. Let _additionalFieldsCopy_ be OrdinaryObjectCreate(%Object.prototype%).
+            1. Let _additionalFieldsCopy_ be OrdinaryObjectCreate(*null*).
             1. Perform ? CopyDataProperties(_additionalFieldsCopy_, _additionalFields_, « », « *undefined* »).
-            1. Return CalendarDateMergeFields(_calendar_.[[Identifier]], _fieldsCopy_, _additionalFieldsCopy_).
+            1. NOTE: Every property of _additionalFieldsCopy_ is a data property with non-*undefined* value, but some property keys may be Symbols.
+            1. Let _excludedKeys_ be an empty Set.
+            1. Let _additionalKeys_ be ! _additionalFieldsCopy_.[[OwnPropertyKeys]]().
+            1. For each element _key_ of _additionalKeys_, do
+              1. Add _key_ to _excludedKeys_.
+              1. If _calendar_.[[Identifier]] is *"iso8601"*, then
+                1. If _key_ is *"month"*, add *"monthCode"* to _excludedKeys_.
+                1. Else if _key_ is *"monthCode"*, add *"month"* to _excludedKeys_.
+              1. Else,
+                1. Let _toExclude_ be CalendarKeysToExclude(_calendar_, _key_).
+                1. Set _excludedKeys_ to the union of _excludedKeys_ and the set of _toExclude_'s elements.
+            1. Let _merged_ be OrdinaryObjectCreate(%Object.prototype%).
+            1. Let _excludedKeysList_ be the List of _excludedKeys_' elements.
+            1. Perform ? CopyDataProperties(_merged_, _fields_, _excludedKeysList_, « *undefined* »).
```

Comment:
It's worth noting that this observably prevents reading "month"/"monthCode"/etc. properties from a user-provided _fields_ object when _additionalFields_ includes covering fields, in contrast with the normal pattern of "observably copy all fields". Should this instead follow the normal pattern and then unobservably remove overridden fields that didn't come from _additionalFields_?

(and likewise for DefaultMergeCalendarFields)

Reply:
I thought that [your earlier comment](https://github.com/tc39/proposal-temporal/pull/2474#discussion_r1071804934) had asked for it to be done this way, but looking back at it I now can't understand how I thought that. Sorry.

I could go either way on this. I think it's fine to not read fields if we're already aware we're not going to use them, but reading them was the status quo before this PR, so preserving that would make the scope of this change smaller.